### PR TITLE
Add ignore package settings to mypy

### PR DESCRIPTION
**Pull request by z103cb**
Added setings to ignore a list of modules for which there no type hints available.

### CODE REVIEW

File: src/caikit_nlp_client/grpc_client_introspection.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def generate_text(self, model_id: str, text: str, **kwargs) -> str:
             log.error(f"Caught exception {exc}, re-throwing")
             raise exc
 
-    def generate_text_stream(self, model_id: str, text: str, **kwargs) -> [str]:
+    def generate_text_stream(self, model_id: str, text: str, **kwargs) -> list[str]:
```
Comment (dtrifiro) [inline src/caikit_nlp_client/grpc_client_introspection.py:92]: This is invalid in python 3.8 (`TypeError: 'type' object is not subscriptable`), so, depending on what version we want to target, we might want to change this to `List[str]` (making sure to add `from typing import List`)

Comment (z103cb) [reply src/caikit_nlp_client/grpc_client_introspection.py:92]: I have tried the suggestion. But it seems that `ruff -format` is modifing this line to look like this. Is there away to disable this behaviour in `ruff` ?

Comment (z103cb) [reply src/caikit_nlp_client/grpc_client_introspection.py:92]: Also, we don't seem to support python 3.8 in the project as defined in the tool. AFAIK python 3.8 has reached EOL.

Comment (dtrifiro) [reply src/caikit_nlp_client/grpc_client_introspection.py:92]: Let's drop 3.8.

We can set the target version in `ruff` using `target-version` in the `[too.ruff]` section, see [`target-version`](https://docs.astral.sh/ruff/settings/#target-version) docs

**Review by dtrifiro — APPROVED** [review]

**Pull request merged**